src/StripedBackupSet.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace AgDatabaseMove
+{
+
+  public class StripedBackupSet
+  {
+    public IEnumerable<BackupMetadata> StripedBackups { get; private set; }
+
+    public StripedBackupSet(IEnumerable<BackupMetadata> backups, BackupMetadata backupToMatch)
+    {
+      var stripes = backups.Where(backup => AreTwoBackupsStriped(backup, backupToMatch));
+      StripedBackups = new List<BackupMetadata>(stripes);
+    }
+
+    public static bool AreTwoBackupsStriped(BackupMetadata backup, BackupMetadata backupToMatch)
+    {
+      return new BackupMetadataEqualityComparer().EqualsExceptForPhysicalDeviceName(backup, backupToMatch);
+    }
+
+    public static IEnumerable<StripedBackupSet> GetStripedBackupSetChain(IEnumerable<BackupMetadata> backups)
+    {
+      List<StripedBackupSet> chain = new List<StripedBackupSet>(); 
+      while(backups.Count() > 0)
+      {
+        var backup = backups.First();
+        var stripedBackupSet = new StripedBackupSet(backups, backup);
+        chain.Add(stripedBackupSet);
+        backups = backups.Except(stripedBackupSet.StripedBackups, new BackupMetadataEqualityComparer());
+      }
```

Comment:
Outer loop here makes sense... but I don't care for removing items from the list and passing the list around constantly (copying data).
Why don't you order the backup list, I think it might already be ordered....
With ordering any striped backup is always going to be the next element... roll them up until you find a non-matching `BackupMetadata`.  I think this can be accomplished with a LINQ GroupBy clause

Reply:
Ya actually `GroupBy` might be best. Then we just have to change those groups into `StripedBackupSet`
Will take a look